armi/bookkeeping/memoryProfiler.py

```diff
@@ -101,11 +129,30 @@ def interactEveryNode(self, cycle, node):
             mpiAction.broadcast().invoke(self.o, self.r, self.cs)
 
     def interactEOL(self):
-        r"""End of life hook. Good place to wrap up or print out summary outputs."""
+        """End of life hook. Good place to wrap up or print out summary outputs."""
         if self.cs["debugMem"]:
```

Comment:
I'm just wondering... should this same settings check be done at `interactBOL` and `interactEveryNode`?

I could be totally off-base here, I just can't tell why this is only done at the one time node.

Reply:
I'm not following. Can you elaborate?

Reply:
Well, this memory profiler does nothing in `interactEOL` unless the setting `debugMem` is set to `True`.

So.... I would EXPECT that setting would control all the time points, not just the one. 

If that's not the case cool, I'm just wondering what the difference is.

Reply:
That setting does get used in `interactBOL` and `interactEveryNode`. These two just also do other stuff too. 

Granted I don't know the original intent. The git blame for the vast majority of this module go well beyond my time with TerraPower.